src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java

```
@@ -240,10 +240,147 @@ private void addRole(String type, Role role) {
+    String overwrite = request.getParameter("overwrite");
+    
+    if (type != null && roleName != null && permissions != null) {
+        boolean overwriteb = false;
+        String pattern = ".*";
+
+        if (type != RoleBasedAuthorizationStrategy.GLOBAL && request.getParameter("pattern") != null) {
+            pattern = request.getParameter("pattern");
+        }
+        
+        if (overwrite != null) {
+            if (overwrite.equals("true")) {
+                overwriteb = true;
+            }
+        }
+        
+        ArrayList<String> permissionList = new ArrayList<String>(); 
+        String[] split = permissions.split(",");
+        for (int i=0; i<split.length; i++) {
+            permissionList.add(split[i]);
+        }
+        
+        Set<Permission> permissionSet = new HashSet<Permission>();    
+        for (String p: permissionList) {
+          permissionSet.add(Permission.fromId(p));
+        }
+        Role role = new Role(roleName, pattern, permissionSet);
+        if (overwriteb) {
+          Role role2 = this.grantedRoles.get(type).getRole(roleName);
+          if (role2!=null) {
+              this.grantedRoles.get(type).removeRole(role2);
+          }
+        }
+        addRole(type, role); 
+        Hudson.getInstance().save();
+    }   
+  }
+    
+  /**
+   * API method to remove roles
+   * @param request has to contain the following parameters
+   *    type (globalRoles, projectRoles)
+   *    roleNames
+   * example: curl -X POST localhost:8080/role-strategy/strategy/removeRoles --data "type=globalRoles&amp;roleNames=ADM,DEV"
+   * @param response
+   * @throws IOException
+   */
+  public void doRemoveRoles(StaplerRequest request, StaplerResponse response) throws IOException {
+    Hudson.getInstance().checkPermission(Jenkins.ADMINISTER);
+    String type = request.getParameter("type");
+    String roleNames = request.getParameter("roleNames");
+    
+    if (type != null && roleNames != null) {
```

The operation should fail with error if parameters are not specified